models/lenet/lenet.hpp

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file lenet.hpp
+ * @author Eugene Freyman
+ * @author Daivik Nema
+ * @author Kartik Dutt
+ * 
+ * Definition of LeNet generally used for object detection.
+ * 
+ * For more information, kindly refer to the following paper.
+ * 
+ * @code
+ * @article{LeCun1998,
+ *  author = {Yann LeCun, Leon Bottou, Yoshua Bengio, Pattrick Haffner},
+ *  title = {Gradient Based Learning Applied to Document Recognizition},
+ *  journal = {IEEE},
+ *  year = {1998},
+ *  url = {http://yann.lecun.com/exdb/publis/pdf/lecun-01a.pdf}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MODELS_LENET_HPP
+#define MODELS_LENET_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */{
+
+/**
+ * Definition of a LeNet CNN.
+ * 
+ * @param leNetVer Version of LeNet.
+ */
+template<size_t leNetVer = 1>
+class LeNet
+{
+ public:
+  //! Create the LeNet object.
+  LeNet();
+
+  /**
+   * LeNet constructor intializes input shape and number of classes.
+   *
+   * @param inputChannels Number of input channels of the input image.
+   * @param inputWidth Width of the input image.
+   * @param inputHeight Height of the input image.
+   * @param numClasses Optional number of classes to classify images into,
+   *                   only to be specified if includeTop is  true.
+   * @param weights One of 'none', 'imagenet'(pre-training on mnist) or path to weights.
+   */
+  LeNet(const size_t inputChannel,
+        const size_t inputWidth,
+        const size_t inputHeight,
+        const size_t numClasses = 1000,
+        const std::string &weights = "none");
```

Comment:
```suggestion
        const std::string& weights = "none");
```
Tiny style comment. :)